hca/loader.py

```diff
@@ -0,0 +1,100 @@
+        entity_type = entity.identifier.entity_type
+        if entity_type == 'biomaterials' or entity_type == 'files':
+            link_type = 'processes'
+            self.__map_related_entities(submission, entity, link_type, 'inputToProcesses')
+            self.__map_related_entities(submission, entity, link_type, 'derivedByProcesses')
+        elif entity_type == 'processes':
+            self.__map_related_entities(submission, entity, 'protocols')
+            link_type = 'biomaterials'
+            self.__map_related_entities(submission, entity, link_type, 'inputBiomaterials')
+            self.__map_related_entities(submission, entity, link_type, 'derivedBiomaterials')
+            link_type = 'files'
+            self.__map_related_entities(submission, entity, link_type, 'inputFiles')
+            self.__map_related_entities(submission, entity, link_type, 'derivedFiles')
+
+    def __map_related_entities(self, submission: HcaSubmission, entity: Entity, related_entity_type: str, link_name: str = None):
+        if not link_name:
+            link_name = related_entity_type
+        related_entities = self.ingest.get_related_entities(link_name, entity.attributes, related_entity_type)
+        for entity_attributes in related_entities:
+            in_cache = submission.contains_entity(entity_attributes)
+            related_entity = submission.map_ingest_entity(entity_attributes)
+            if not in_cache:
+                self.__add_related_entities(submission, related_entity)
+            submission.link_entities(entity, related_entity)
+
+    def __map_manifest_content(self, submission: HcaSubmission, manifest: Entity):
+        for project_uuid in manifest.attributes.get('fileProjectMap', {}).keys():
+            project = self.__map_entity(submission, 'projects', project_uuid)
+            submission.link_entities(manifest, project)
+
+        for biomaterial_uuid in manifest.attributes.get('fileBiomaterialMap', {}).keys():
+            biomaterial = self.__map_entity(submission, 'biomaterials', biomaterial_uuid)
+            submission.link_entities(manifest, biomaterial)
+
+        for protocol_uuid in manifest.attributes.get('fileProtocolMap', {}).keys():
+            protocol = self.__map_entity(submission, 'protocols', protocol_uuid)
+            submission.link_entities(manifest, protocol)
+
+        for process_uuid in manifest.attributes.get('fileProcessMap', {}).keys():
+            process = self.__map_entity(submission, 'processes', process_uuid)
+            submission.link_entities(manifest, process)
+
+        data_file_map = {}
+        for file_uuid in manifest.attributes.get('fileFilesMap', {}).keys():
+            file = self.__map_entity(submission, 'files', file_uuid)
+            submission.link_entities(manifest, file)
+            data_file_uuid = file.attributes.get('dataFileUuid', '')
+            data_file_map[data_file_uuid] = file
+
+# for data_file_uuid in manifest.attributes.get('dataFiles', []):
```

Comment:
There is no need for comments in the code.
If this is not used - delete. If this should be used - uncomment and fix.

Reply:
I take your point, but this is still a draft PR is not be ready for review.
This code **will** be needed when we map data files for brokering to the ENA, so I'm glad we can return to this commit to retrieve it when we need that functionality, even if this block is removed when we are ready to merge.